test/app/controllers/albums.post.test.js

```diff
@@ -0,0 +1,99 @@
+const request = require('supertest');
+const { factory } = require('factory-girl');
+
+const { resHashedPasswordMock, resComparePasswordMock } = require('../../mocks/bcrypt');
+const { albumResValueMock, albumRejValueMock } = require('../../mocks/albums');
+const app = require('../../../app');
+const { factoryByModel } = require('../../factory/factory_by_models');
+const { albumsErrorMessages } = require('../../errors/albums');
+
+factoryByModel('users');
+
+describe('POST /albums/:id', () => {
+  let mockedPassword = {};
+  let responseKeys = {};
+  let successResponse = {};
+  let existingAlbumResponse = {};
+  let albumNotFoundResponse = {};
+  let adminUser = {};
+  let token = {};
+  let successAlbumId = {};
+  let albumCreationErrorCode = {};
+  let albumNotFoundErrorCode = {};
+  let successResponseBody = {};
+  beforeAll(async () => {
+    await resComparePasswordMock();
+    mockedPassword = await resHashedPasswordMock('adminPass60');
+    responseKeys = ['id', 'albumId', 'title', 'userId', 'updatedAt', 'createdAt', 'deleted_at'];
+    adminUser = await factory.create('users', {
+      firstName: 'Alberto',
+      lastName: 'Alvarez',
+      password: mockedPassword,
+      email: 'admin@wolox.com.ar',
+      admin: true
+    });
+    token = await request(app)
+      .post('/users/sessions')
+      .send({
+        email: 'admin@wolox.com.ar',
+        password: 'adminPass60'
+      })
+      .then(response => response.body.token);
+    successAlbumId = 1;
+    successResponseBody = await {
+      userId: 1,
+      id: 1,
+      title: 'mocked book'
+    };
+    await albumResValueMock(successResponseBody);
+    successResponse = await request(app)
+      .post(`/albums/${successAlbumId}`)
+      .set('Authorization', token);
+    existingAlbumResponse = await request(app)
+      .post(`/albums/${successAlbumId}`)
+      .set('Authorization', token);
+    await albumRejValueMock({});
+    albumNotFoundResponse = await request(app)
+      .post('/albums/500')
+      .set('Authorization', token);
+    albumCreationErrorCode = 'album_creation_error';
+    albumNotFoundErrorCode = 'api_albums_error';
+  });
+  describe('Successful case', () => {
+    describe('Creates a new album for the authenticated user', () => {
+      it('should have status code 201', () => {
+        expect(successResponse.statusCode).toEqual(201);
+      });
+      it('should have the proper keys', () => {
+        expect(Object.keys(successResponse.body)).toEqual(responseKeys);
+      });
+      it('should belong to the user that was authenticated', () => {
+        expect(successResponse.body.userId).toEqual(adminUser.id);
+      });
+    });
+  });
+  describe('Failure cases', () => {
+    describe('Does not create a new album if the user already bought it', () => {
+      it('should have status code 500', () => {
```

Comment:
I think if the user already bought the album, the status code must be in the range of 400